app/script/event/EventRepository.js

```diff
@@ -647,7 +647,8 @@ z.event.EventRepository = class EventRepository {
    * @returns {Promise} Resolves with the saved event
    */
   _handleEventSaving(event, source) {
-    const {conversation: conversationId, id: eventId} = event;
+    const conversationId = event.conversation;
+    const eventId = (event.data || {}).replacing_message_id || event.id;
```

Comment:
Let's move this back to the object destructuring so we can set a default there:

````
const {conversation: conversationId, data: eventData = {}} = event;
const eventId = eventData.replacing_message_id || event.id;
````